precompiles/staking/types.go

```diff
@@ -49,6 +55,96 @@ type EventCancelUnbonding struct {
 	CreationHeight   *big.Int
 }
 
+// NewMsgCreateValidator creates a new MsgCreateValidator instance and does sanity checks
+// on the given arguments before populating the message.
+func NewMsgCreateValidator(args []interface{}, denom string) (*stakingtypes.MsgCreateValidator, common.Address, error) {
+	if len(args) != 7 {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 7, len(args))
+	}
+
+	description, ok := args[0].(struct {
+		Moniker         string "json:\"moniker\""
+		Identity        string "json:\"identity\""
+		Website         string "json:\"website\""
+		SecurityContact string "json:\"securityContact\""
+		Details         string "json:\"details\""
+	})
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidType, "description", "tuple", args[0])
+	}
+
+	// since solidity does not support decimals, after passing in the big int, convert the big int into a decimal with a precision of 18
+	commission, ok := args[1].(struct {
+		Rate          *big.Int "json:\"rate\""
+		MaxRate       *big.Int "json:\"maxRate\""
+		MaxChangeRate *big.Int "json:\"maxChangeRate\""
```

Comment:
ditto: define a struct and use `Copy` method, check how `distribution` precompile types does it, something like this:
```go
var input ValidatorSlashesInput
	if err := method.Inputs.Copy(&input, args); err != nil {
		return nil, fmt.Errorf("error while unpacking args to ValidatorSlashesInput struct: %s", err)
	}
```

Reply:
I see that other functions such as **NewMsgDelegate**, **NewMsgUndelegate**, etc. do not use this check. Only some request parameters such as **NewValidatorSlashesRequest** have this check method. Is it necessary?